### DISCUSSION TITLE
How to set the random seed for curobo?

### DISCUSSION THREAD

**Discussion by TooSchoolForCool** [NVlabs/curobo#200] [Software Q&A]
How can I set the random seed for curobo?

I have try following global setting method
```python
seed = 123
torch.manual_seed(seed)
np.random.seed(seed)
random.seed(seed)
```
But none of them actually work.

Every time I run follow script, it always give me same results.
```python
ik_solver = IKSolver(xxx)

q_samples = ik_solver.sample_configs(10)
```
`q_samples` are always the same each time I run this script.

**Comment by balakumar-s** [comment]
It's hard coded here: https://github.com/NVlabs/curobo/blob/b481ee201a2754c072ba9bcd80bd5d327fddc599/src/curobo/wrap/reacher/ik_solver.py#L337

We can expose it as a parameter in the next release.

**Reply by TooSchoolForCool** [reply]
Thank you so much!

I will expose it at my side first :).

**Reply by TooSchoolForCool** [reply]
Hi Balakumar,

I discovered that changing the random seed does not affect the behavior of the `self.sample_configs()` function because it samples joint configurations from `self.rollout_fn.sample_random_actions` instead of `self.q_sample_gen`.

I've submitted a [pull request](https://github.com/NVlabs/curobo/pull/202) that exposes `random_seed` and updates the `sample_configs` method for your reference.

**Reply by balakumar-s** [reply]
Thanks, this has been added: https://github.com/NVlabs/curobo/blob/774dcfd609847ad98147f166ea2f5954516ff7a0/src/curobo/wrap/reacher/ik_solver.py#L98
